src/routes/spellbook/+page.svelte

```diff
@@ -78,9 +101,52 @@
             addSpell(spells[0]);
             return;
         }
+        await searchSpells();
+    }
 
+    async function handleLevelUpdate(event: CustomEvent) {
+        selectedLevel = event.detail.level;
         await searchSpells();
     }
+
+    async function handleClassUpdate(event: CustomEvent) {
+        selectedClass = event.detail.class;
+        await searchSpells();
+    }
+
+
+    function handleLevelSort() {
+        if(isSortingByAscending){
+            spells = spells.sort((a, b) => b.level_int - a.level_int);
+            isSortingByAscending=false;
+        }
+        else{
+            spells = spells.sort((a, b) => a.level_int - b.level_int);
+            isSortingByAscending=true;
+        }
+    }
+
+    function handleNameSort() {
+        spells = [...originalSpellsOrder].sort((a,b)=> {
+            return a.name.localeCompare(b.name);
+        });
+    }
+
+    function groupSpellsByLevel() {
+        const currentSpellBook = $spellbookStore[$activeSpellbookIndex];
+        if(currentSpellBook){
+            groupedSpells = {};
+
+            currentSpellBook.spells.forEach(s => {
+            const levelKey = s.level_int < 1 ? "Cantrips" : `Level ${s.level_int} spells`;
+            if (!groupedSpells[levelKey]) {
+                groupedSpells[levelKey] = [];
+            }
+            groupedSpells[levelKey].push(s);
+        });
+        }
```

Comment:
Statement kan inverteres for at undgå nesting.

```suggestion
        const currentSpellBook = $spellbookStore[$activeSpellbookIndex];
        if (!currentSpellBook) {
            console.log("No spellbook found based on index", $activeSpellbookIndex);
            return;
        }

        groupedSpells = {};
        currentSpellBook.spells.forEach(s => {
            const levelKey = s.level_int < 1 ? "Cantrips" : `Level ${s.level_int} spells`;
            if (!groupedSpells[levelKey]) {
                groupedSpells[levelKey] = [];
            }
            groupedSpells[levelKey].push(s);
        });
```